gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/store/GemFireContainer.java

```diff
@@ -6510,8 +6510,11 @@ public void accountIndexMemory(boolean askMemoryManager, boolean isDestroy) {
         if (askMemoryManager) {
           // Only acquire memory while initial index creation. Rest all index accounting will be done by
           // region put/delete
-          Misc.getCacheLogWriter().info("Total overhead computed="+ sum + "intialAccounting="+intialAccounting);
-          baseRegion.acquirePoolMemory(0, sum - intialAccounting,
+          Misc.getCacheLogWriter().info("Total overhead computed = "+ sum + " intialAccounting="+intialAccounting);
+          // Negative value means index is being recreated. Kind of hackish but safe and wont stop server from crashing due to
```

Comment:
If you are clear about the situation of recreation of index then you can just clear the 'intialAccounting' field to 0 in 'clearallIndexes' method, which is called just before recreating the index.